tortoisebeacon/beacon_calc.go

```diff
@@ -1,32 +1,35 @@
 package tortoisebeacon
 
 import (
+	"context"
 	"fmt"
-	"sort"
 	"strings"
 
 	"github.com/spacemeshos/go-spacemesh/common/types"
 	"github.com/spacemeshos/go-spacemesh/events"
 	"github.com/spacemeshos/go-spacemesh/log"
 )
 
-func (tb *TortoiseBeacon) calcBeacon(epoch types.EpochID, coinflip bool) error {
+func (tb *TortoiseBeacon) calcBeacon(ctx context.Context, epoch types.EpochID, lastRoundVotes allVotes) error {
+	select {
```

Comment:
imo this is unnecessary. calcBeacon is just a computation and suppose not to take long time, right?

Reply:
It is just computation, but I made `runConsensusPhase` return an empty list of votes if the context is cancelled, so I need to check it in `calcBeacon` too and return in this case. But I think it's better to indicate the cancelled context explicitly. What I have now is not clear